libraries/ms-common/inc/usart.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+// USART Interface
+#include <stdint.h>
+
+#include "status.h"
+#include "interrupt.h"
+#include "gpio.h"
+
+// For setting word length
+typedef enum {
+	USART_WORD_LENGTH_8b = 0,
+	USART_WORD_LENGTH_9b, 
+	USART_WORD_LENGTH_7b,
+} USARTWordLength;
+
+// For setting number of stop bits
+typedef enum {
+	USART_STOP_BITS_1 = 0,
+	USART_STOP_BITS_2,
+	USART_STOP_BITS_1_5,
+} USARTStopBits;
+
+// For setting parity mode
+typedef enum {
+	USART_PARITY_NO = 0,
+	USART_PARITY_EVEN,
+	USART_PARITY_ODD,
+} USARTParity;
+
+// For setting receive or transmit mode
+typedef enum {
+	USART_MODE_RX = 0,
+	USART_MODE_TX,
+} USARTMode;
+
+// For setting hardware flow control mode
+typedef enum {
+	USART_HARDWARE_FLOW_CONTROL_NONE = 0,
+	USART_HARDWARE_FLOW_CONTROL_RTS,
+	USART_HARDWARE_FLOW_CONTROL_CTS,
+	USART_HARDWARE_FLOW_CONTROL_RTS_CTS,
+} USARTHardwareFlowControl;
```

Comment:
If you're supporting Hardware Flow Control, I think you'll also need to initialize the CTS pin in your ``USARTSettings``.

(Not sure if we'll use this functionality though)